src/components/toast/toast.component.js

```diff
@@ -128,6 +134,10 @@ const Toast = React.forwardRef(
             {...tagComponent(restProps["data-component"] || "toast", restProps)}
             {...toastProps}
             ref={toastContentNodeRef}
+            tabIndex={-1}
+            role="status"
+            aria-atomic="true"
+            onKeyDown={handleKeyDown}
```

Comment:
I'm curious why we need a custom `onKeyDown` here to focus the close icon when tabbing from the container. For one thing, this should work automatically - I assume it doesn't since you've added this, but I'd rather figure out what the underlying problem is with that and solve it rather than just hardcoding the specific behaviour we need.

Especially because this onKeyDown is now on the whole toast container so will it not mean tabbing from anywhere with in will focus the close icon even when it isn't supposed to? Eg if you're already focused on the close icon won't this then refocus it, therefore stopping keyboard users navigating away...

Reply:
This was added because when `Toast` is re-opened from a `Dialog` [like in this example](https://carbon.sage.com/?path=/docs/dialog--other-focusable-containers), the close icon would be the first item to get focus. I couldn't get it to work any other way than this but I'm open to other potential fixes if you have any?